src/Endpoints/Indexes.php

```diff
@@ -145,26 +145,33 @@ public function waitForPendingUpdate($updateId, $timeoutInMs = 5000, $intervalIn
      */
     public function search($query, array $searchParams = [], array $options = [])
     {
-        $parameters = array_merge(
-            ['q' => $query],
-            $searchParams
-        );
-
-        $result = $this->http->post(self::PATH.'/'.$this->uid.'/search', $parameters);
+        $result = $this->rawSearch($query, $searchParams);
 
         if (\array_key_exists('raw', $options) && $options['raw']) {
             return $result;
         }
 
         $searchResult = new SearchResult($result);
-
-        if (\array_key_exists('transformHits', $options) && \is_callable($options['transformHits'])) {
-            $searchResult = $searchResult->filter($options['transformHits']);
-        }
+        $searchResult = $searchResult->applyOptions($options);
```

Comment:
If `applyOptions` is always used after `new SearchResult`, should it not be in the constructor of `SearchResult` ?

Reply:
Also, are modifications not apply directly on the instance `SearchResult`? I'm not sure ` $searchResult =` is needed.

Reply:
I'm not sure I want to pass it in the constructor. It's not because you create a `SearchResult()` you want automatically the options to be applied.
This class is public so our users can use it in their own ways.
I agree, catch the return is not mandatory.

Reply:
```suggestion
      $searchResult->applyOptions($options);
```